packages/components/src/Button/button.tsx

```diff
@@ -3,19 +3,19 @@ import React, { ReactNode } from "react";
 
 type ButtonHTMLElementProps = React.ButtonHTMLAttributes<HTMLButtonElement>;
 
-export type ButtonProps = {
+type ButtonPropsBase = ButtonHTMLElementProps & {
   as?: "button" | React.ComponentType<any>;
   /**
    * The button contents or label.
    */
   children: ReactNode;
   color?: ClickableProps<"button">["color"];
-  disabled?: ButtonHTMLElementProps["disabled"];
```

Comment:
I removed `disabled` and `type` from this list since we're now using everything in `ButtonHTMLElementProps`. Without combining `ButtonHTMLElementProps`, the other attributes we want (`data-testid`, `role`, `aria-live`, `tabIndex`) freak flow out.

Reply:
> Without combining ButtonHTMLElementProps, the other attributes we want (data-testid, role, aria-live, tabIndex) freak flow out.

Do you mean that we can't add those attributes without also combining `ButtonHTMLElementProps`? That seems odd 🤔

Reply:
This is the test output before this change when testing on a button that has `data-testid`, `role`, `aria-live`, and `tabIndex`:
```
src/Button/button.tsx(48,5): error TS2322: Type '{ children: string; "data-testid": string; role: string; "aria-live": string; tabIndex: number; }' is not assignable to type 'IntrinsicAttributes & Pick<ButtonPropsBase, "as" | "children" | "color" | "size" | "variant" | "disabled" | "type">'.
  Property 'role' does not exist on type 'IntrinsicAttributes & Pick<ButtonPropsBase, "as" | "children" | "color" | "size" | "variant" | "disabled" | "type">'.
  ```

Reply:
Weird. I can't repro that, but was probably testing something different.

In any case, intersecting with `ButtonHTMLElementProps` here seems good 👍